### PR TITLE
Group delete related changes

### PR DESCRIPTION
**Pull request by pkdash**
This PR consists of the following changes:
- Removed group delete option from group edit modal dialog
- Added separate delete/restore buttons on group page
- If there are deleted groups, then owners can display/hide (default is hide) those groups in my groups page.

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi or @zhiyuli can you review this small PR? This needs to go in for AGU release.

**Comment by pkdash** [comment]
Just realized couple of unit tests related to group will be failing. Will be fixing soon.

**Comment by hyi** [comment]
@pkdash I am done with first round of review.

**Comment by pkdash** [comment]
@hyi Do you have any comments for me to address?

**Comment by hyi** [comment]
@pkdash I have several comments for you to address. I used github's new review format to post my review comments, and it appears you did not receive them. Anyway, I have clicked "submit review" button just now again, you should be able to see all my review comments to be addressed for this PR dirctly on github page if you don't receive email notifications. Let me know if you have trouble seeing my review comments.

File: hs_core/templates/pages/my-groups.html [hunk 1]
```diff
@@ -140,12 +151,50 @@ <h2 class="page-title">My Groups</h2>
...
+                                            {% else %}
+                                                <div class="group-image-wrapper">
+                                                    <div class="group-image-small"
+                                                         style="background-image: url({{ STATIC_URL }}img/home-page/step4.png)"></div>
+                                                </div>
+                                            {% endif %}
+                                        </td>
+
+                                        <td>
+                                            <h4><a href="/group/{{ group.id }}" style="color: red">{{ group.name }}</a></h4>
```
Comment (pkdash) [inline hs_core/templates/pages/my-groups.html]: Have shown it to @dtarb . Without the red color there would be no way to know which ones at inactive groups. I guess there are other ways to differential though.

Comment (pkdash) [inline hs_core/templates/pages/my-groups.html]: I wanted to deploy it then and no deployment server was available at that time.

File: hs_core/views/__init__.py [hunk 2]
```diff
@@ -946,6 +938,34 @@ def update_user_group(request, group_id, *args, **kwargs):
...
+    action = 'delete' if delete_or_restore.lower() == 'delete' else 'restore'
+    if user.uaccess.can_change_group_flags(group):
+        group.gaccess.active = status
+        group.gaccess.save()
+        messages.success(request, "Group {} was successful.".format(action))
+    else:
+        messages.error(request, "Group {} errors: You don't have permission to {}"
+                                " this group.".format(action, action))
+
+    return HttpResponseRedirect(request.META['HTTP_REFERER'])
```
Comment (pkdash) [inline hs_core/views/__init__.py]: I made it an utility function. But I won't be surprised if the REST api does not use it.

**Comment by pkdash** [comment]
@hyi I have now addressed your comments.

**Comment by hyi** [comment]
+1

**Comment by pkdash** [comment]
Will be adding a unit test for the utility function I added before merging.

**Comment by pkdash** [comment]
@hyi Once you review the added unit test code I can merge it.

**Comment by hyi** [comment]
@pkdash You added unit test code looks good. You can merge it.

**Comment by pkdash** [comment]
@hyi Thanks Hong.

**Pull request merged**